researchstack-sdk/src/main/java/org/sagebionetworks/bridge/researchstack/BridgeDataProvider2.java

```diff
@@ -0,0 +1,285 @@
+package org.sagebionetworks.bridge.researchstack;
+
+import android.content.Context;
+import android.support.annotation.AnyThread;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.researchstack.backbone.DataProvider;
+import org.researchstack.backbone.DataResponse;
+import org.researchstack.backbone.model.ConsentSignatureBody;
+import org.researchstack.backbone.model.SchedulesAndTasksModel;
+import org.researchstack.backbone.model.User;
+import org.researchstack.backbone.result.TaskResult;
+import org.researchstack.backbone.task.Task;
+import org.sagebionetworks.bridge.android.manager.BridgeManagerProvider;
+import org.sagebionetworks.bridge.android.manager.auth.AuthenticationManager;
+import org.sagebionetworks.bridge.rest.model.SharingScope;
+import org.sagebionetworks.bridge.rest.model.StudyParticipant;
+import org.sagebionetworks.bridge.rest.model.UserSessionInfo;
+
+import rx.Completable;
+import rx.Observable;
+import rx.Observer;
+import rx.functions.Action0;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.sagebionetworks.bridge.researchstack.ApiUtils.SUCCESS_DATA_RESPONSE;
+
+/**
+ * Created by jyliu on 1/20/2017.
+ */
+@AnyThread
+public class BridgeDataProvider2 extends DataProvider {
+    private final BridgeManagerProvider bridgeManagerProvider;
+
+    public BridgeDataProvider2(BridgeManagerProvider bridgeManagerProvider) {
+        this.bridgeManagerProvider = bridgeManagerProvider;
+    }
+
+    @Override
+    @Deprecated
+    public Observable<DataResponse> initialize(@NonNull Context context) {
+        checkNotNull(context);
+
+        return Observable.just(new DataResponse(true, "donothing"));
+    }
+
+    @Override
+    @NonNull
+    @Deprecated
+    public Observable<DataResponse> signUp(@NonNull Context context, @NonNull String email,
+                                           @NonNull String username, @NonNull String password) {
+        return signUp(email, password);
+    }
+
+    @NonNull
+    public Observable<DataResponse> signUp(@NonNull String email, @NonNull String password) {
+        checkNotNull(email);
+        checkNotNull(password);
+
+        return bridgeManagerProvider.getAuthenticationManager()
```

Comment:
move bridgeManagerProvider.getAuthenticationManager() to helper method called getAuthManager()?